File: 696-count-binary-substrings/count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        previousGroup = 0
        currentGroup = 1
        count = 0
        for i in range(len(s)-1):
            if s[i]==s[i+1]:
                currentGroup +=1
            else:
                previousGroup = currentGroup
                currentGroup = 1
            if previousGroup >= currentGroup:
                count +=1
        return count
        

File: 696-count-binary-substrings/count-binary-substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int prevGroup = 0, currGroup = 1, count = 0;
        for(int i = 0; i<s.length()-1; ++i){
            if(s[i] == s[i+1]){
                ++currGroup;
            }
            else{
                prevGroup = currGroup;
                currGroup = 1;
            }
            if(prevGroup >= currGroup){
                ++count;
            }
        }
        return count;
    }
};